libs/cardano-ledger-core/testlib/Test/Cardano/Ledger/Core/Arbitrary.hs

```diff
@@ -633,12 +634,33 @@ uniformSubset mSubSetSize inputSet gen = do
       | i <= 0 = pure acc
       | otherwise = do
           ix <- uniformRM (0, Set.size s - 1) gen
-          go (Set.insert (Set.elemAt ix s) acc) (Set.deleteAt ix s) (i - 1)
+          go (Set.deleteAt ix s) (Set.insert (Set.elemAt ix s) acc) (i - 1)
```

Comment:
Kind of, half the time at least :smile: 

Depending if the requested size was even or odd it would either produce an empty Set or a the full original Set with a single random element removed from it.